doc/sphinx/source/graph-basics.rst

```diff
@@ -117,8 +185,36 @@ corresponding to the array provided as the nth argument, and the outputs
 of all result ops will be written into the result arrays in row-major
 order.
 
+
+Shared pointers
```

Comment:
I think this section is probably best left where it was originally. Shared pointers are not a fundamental concept to nGraph, just an implementation detail of how graphs are built in the C++ API.